// @mux/mux-node/resources/system/signing-keys.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.SigningKeys = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Signing keys are used to sign JSON Web Tokens (JWTs) for securing certain requests, such as secure playback URLs and access to real-time viewer counts in Mux Data. **One signing key can be used to sign multiple requests - you probably only need one active at a time.** However, you can create multiple signing keys to enable key rotation, creating a new key and deleting the old only after any existing signed requests have expired.
 */
class SigningKeys extends resource_1.APIResource {
    /**
     * Creates a new signing key pair. When creating a new signing key, the API will
     * generate a 2048-bit RSA key-pair and return the private key and a generated
     * key-id; the public key will be stored at Mux to validate signed tokens.
     */
    create(options) {
        return this._client.post('/system/v1/signing-keys', {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
    /**
     * Retrieves the details of a signing key that has previously been created. Supply
     * the unique signing key ID that was returned from your previous request, and Mux
     * will return the corresponding signing key information. **The private key is not
     * returned in this response.**
     */
    retrieve(signingKeyID, options) {
        return this._client.get((0, path_1.path) `/system/v1/signing-keys/${signingKeyID}`, {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
    /**
     * Returns a list of signing keys.
     */
    list(query = {}, options) {
        return this._client.getAPIList('/system/v1/signing-keys', (pagination_1.BasePage), {
            query,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        });
    }
    /**
     * Deletes an existing signing key. Use with caution, as this will invalidate any
     * existing signatures and no JWTs can be signed using the key again.
     */
    delete(signingKeyID, options) {
        return this._client.delete((0, path_1.path) `/system/v1/signing-keys/${signingKeyID}`, {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
exports.SigningKeys = SigningKeys;
