// @mux/mux-node/resources/video/playback-restrictions.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { BasePage } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Playback Restrictions allows you to set additional rules for playing videos. You can set the domains/hostnames allowed to play your videos. For instance, viewers can play videos embedded on the `https://example.com` website when you set the Playback Restrictions with `example.com` as an allowed domain. Any Video requests from other websites are denied. Additionally, you can set rules for what user agents to allow. Please see [Using User-Agent HTTP header for validation](https://docs.mux.com/guides/secure-video-playback#using-user-agent-http-header-for-validation) for more details on this feature.
 */
export class PlaybackRestrictions extends APIResource {
    /**
     * Create a new Playback Restriction.
     *
     * @example
     * ```ts
     * const playbackRestriction =
     *   await client.video.playbackRestrictions.create({
     *     referrer: {
     *       allowed_domains: ['*.example.com'],
     *       allow_no_referrer: true,
     *     },
     *     user_agent: {
     *       allow_no_user_agent: false,
     *       allow_high_risk_user_agent: false,
     *     },
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/video/v1/playback-restrictions', {
            body,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
    /**
     * Retrieves a Playback Restriction associated with the unique identifier.
     *
     * @example
     * ```ts
     * const playbackRestriction =
     *   await client.video.playbackRestrictions.retrieve(
     *     'PLAYBACK_RESTRICTION_ID',
     *   );
     * ```
     */
    retrieve(playbackRestrictionID, options) {
        return this._client.get(path `/video/v1/playback-restrictions/${playbackRestrictionID}`, {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
    /**
     * Returns a list of all Playback Restrictions.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const playbackRestriction of client.video.playbackRestrictions.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/video/v1/playback-restrictions', (BasePage), {
            query,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        });
    }
    /**
     * Deletes a single Playback Restriction.
     *
     * @example
     * ```ts
     * await client.video.playbackRestrictions.delete(
     *   'PLAYBACK_RESTRICTION_ID',
     * );
     * ```
     */
    delete(playbackRestrictionID, options) {
        return this._client.delete(path `/video/v1/playback-restrictions/${playbackRestrictionID}`, {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
    /**
     * Allows you to modify the list of domains or change how Mux validates playback
     * requests without the `Referer` HTTP header. The Referrer restriction fully
     * replaces the old list with this new list of domains.
     *
     * @example
     * ```ts
     * const playbackRestriction =
     *   await client.video.playbackRestrictions.updateReferrer(
     *     'PLAYBACK_RESTRICTION_ID',
     *     {
     *       allowed_domains: ['*.example.com'],
     *       allow_no_referrer: true,
     *     },
     *   );
     * ```
     */
    updateReferrer(playbackRestrictionID, body, options) {
        return this._client.put(path `/video/v1/playback-restrictions/${playbackRestrictionID}/referrer`, {
            body,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
    /**
     * Allows you to modify how Mux validates playback requests with different user
     * agents. Please see
     * [Using User-Agent HTTP header for validation](https://docs.mux.com/guides/secure-video-playback#using-user-agent-http-header-for-validation)
     * for more details on this feature.
     *
     * @example
     * ```ts
     * const playbackRestriction =
     *   await client.video.playbackRestrictions.updateUserAgent(
     *     'PLAYBACK_RESTRICTION_ID',
     *     {
     *       allow_high_risk_user_agent: false,
     *       allow_no_user_agent: false,
     *     },
     *   );
     * ```
     */
    updateUserAgent(playbackRestrictionID, body, options) {
        return this._client.put(path `/video/v1/playback-restrictions/${playbackRestrictionID}/user_agent`, {
            body,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
}
