// @mux/mux-node/resources/video/playback.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Playback = void 0;
const resource_1 = require("../../core/resource.js");
const headers_1 = require("../../internal/headers.js");
const path_1 = require("../../internal/utils/path.js");
class Playback extends resource_1.APIResource {
    /**
     * [Fetch an animated GIF or WebP image](https://docs.mux.com/guides/get-images-from-a-video#get-an-animated-gif-from-a-video)
     * from a video segment with optional transformations.
     *
     * @example
     * ```ts
     * const response = await client.video.playback.animated(
     *   'PLAYBACK_ID',
     *   'gif',
     * );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    animated(playbackId, extension, query = {}, options) {
        return this._client.get((0, path_1.path) `/${playbackId}/animated.${extension}`, {
            query,
            defaultBaseURL: 'https://image.mux.com',
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'image/gif' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Fetch an HLS (HTTP Live Streaming) playlist for the specified video asset, with
     * optional query parameters to
     * [modify playback behavior](https://docs.mux.com/guides/modify-playback-behavior).
     *
     * @example
     * ```ts
     * const response = await client.video.playback.hls(
     *   'PLAYBACK_ID',
     * );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    hls(playbackId, query = {}, options) {
        return this._client.get((0, path_1.path) `/${playbackId}.m3u8`, {
            query,
            defaultBaseURL: 'https://stream.mux.com',
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'application/vnd.apple.mpegurl' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Fetch a static rendition (usually an MP4 or M4A file) of the specified video
     * asset. [MP4 Support](https://docs.mux.com/guides/enable-static-mp4-renditions)
     * must be enabled on the asset before using these URLs.
     *
     * @example
     * ```ts
     * const response =
     *   await client.video.playback.staticRendition(
     *     'PLAYBACK_ID',
     *     'capped-1080p.mp4',
     *   );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    staticRendition(playbackId, filename, query = {}, options) {
        return this._client.get((0, path_1.path) `/${playbackId}/${filename}`, {
            query,
            defaultBaseURL: 'https://stream.mux.com',
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'video/mp4' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Fetch a storyboard image composed of multiple thumbnails for use in
     * [timeline hover previews](https://docs.mux.com/guides/create-timeline-hover-previews).
     *
     * @example
     * ```ts
     * const response = await client.video.playback.storyboard(
     *   'PLAYBACK_ID',
     *   'jpg',
     * );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    storyboard(playbackId, extension, query = {}, options) {
        return this._client.get((0, path_1.path) `/${playbackId}/storyboard.${extension}`, {
            query,
            defaultBaseURL: 'https://image.mux.com',
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'image/jpeg' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Fetch metadata for the
     * [storyboard image in JSON format](https://docs.mux.com/guides/create-timeline-hover-previews#json),
     * detailing the coordinates and time ranges of each thumbnail.
     *
     * @example
     * ```ts
     * const response = await client.video.playback.storyboardMeta(
     *   'PLAYBACK_ID',
     * );
     * ```
     */
    storyboardMeta(playbackId, query = {}, options) {
        return this._client.get((0, path_1.path) `/${playbackId}/storyboard.json`, {
            query,
            defaultBaseURL: 'https://image.mux.com',
            ...options,
        });
    }
    /**
     * Fetch metadata for the
     * [storyboard image in WebVTT format](https://docs.mux.com/guides/create-timeline-hover-previews#webvtt),
     * detailing the coordinates and time ranges of each thumbnail.
     *
     * @example
     * ```ts
     * const response = await client.video.playback.storyboardVtt(
     *   'PLAYBACK_ID',
     * );
     * ```
     */
    storyboardVtt(playbackId, query = {}, options) {
        return this._client.get((0, path_1.path) `/${playbackId}/storyboard.vtt`, {
            query,
            defaultBaseURL: 'https://image.mux.com',
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'text/vtt' }, options?.headers]),
        });
    }
    /**
     * [Fetch a thumbnail image from a video](https://docs.mux.com/guides/get-images-from-a-video)
     * at a specified time with optional transformations.
     *
     * @example
     * ```ts
     * const response = await client.video.playback.thumbnail(
     *   'PLAYBACK_ID',
     *   'jpg',
     * );
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    thumbnail(playbackId, extension, query = {}, options) {
        return this._client.get((0, path_1.path) `/${playbackId}/thumbnail.${extension}`, {
            query,
            defaultBaseURL: 'https://image.mux.com',
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'image/jpeg' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Fetch a standalone WebVTT version of a text track from an asset.
     *
     * @example
     * ```ts
     * const response = await client.video.playback.track(
     *   'PLAYBACK_ID',
     *   'TRACK_ID',
     * );
     * ```
     */
    track(playbackId, trackId, query = {}, options) {
        return this._client.get((0, path_1.path) `/${playbackId}/text/${trackId}.vtt`, {
            query,
            defaultBaseURL: 'https://stream.mux.com',
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'text/vtt' }, options?.headers]),
        });
    }
    /**
     * Fetch a
     * [transcript of an asset](https://docs.mux.com/guides/add-autogenerated-captions-and-use-transcripts#retrieve-a-transcript).
     * This is only possible for assets with a text track generated using the
     * [VOD generated captions feature](https://docs.mux.com/guides/add-autogenerated-captions-and-use-transcripts).
     *
     * @example
     * ```ts
     * const response = await client.video.playback.transcript(
     *   'PLAYBACK_ID',
     *   'TRACK_ID',
     * );
     * ```
     */
    transcript(playbackId, trackId, query = {}, options) {
        return this._client.get((0, path_1.path) `/${playbackId}/text/${trackId}.txt`, {
            query,
            defaultBaseURL: 'https://stream.mux.com',
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'text/plain' }, options?.headers]),
        });
    }
}
exports.Playback = Playback;
