// @mux/mux-node/resources/video/transcription-vocabularies.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import { BasePage } from "../../core/pagination.mjs";
import { buildHeaders } from "../../internal/headers.mjs";
import { path } from "../../internal/utils/path.mjs";
/**
 * Transcription Vocabularies allows you to provide collections of phrases like proper nouns, technical jargon, and uncommon words as part of captioning workflows. When using Auto-Generated Captions, Transcription Vocabularies increase the likelihood of correct speech recognition for these words and phrases.
 */
export class TranscriptionVocabularies extends APIResource {
    /**
     * Create a new Transcription Vocabulary.
     *
     * @example
     * ```ts
     * const transcriptionVocabulary =
     *   await client.video.transcriptionVocabularies.create({
     *     phrases: [
     *       'Mux',
     *       'Live Stream',
     *       'Playback ID',
     *       'video encoding',
     *     ],
     *     name: 'Mux API Vocabulary',
     *   });
     * ```
     */
    create(body, options) {
        return this._client.post('/video/v1/transcription-vocabularies', {
            body,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
    /**
     * Retrieves the details of a Transcription Vocabulary that has previously been
     * created. Supply the unique Transcription Vocabulary ID and Mux will return the
     * corresponding Transcription Vocabulary information. The same information is
     * returned when creating a Transcription Vocabulary.
     *
     * @example
     * ```ts
     * const transcriptionVocabulary =
     *   await client.video.transcriptionVocabularies.retrieve(
     *     'TRANSCRIPTION_VOCABULARY_ID',
     *   );
     * ```
     */
    retrieve(transcriptionVocabularyID, options) {
        return this._client.get(path `/video/v1/transcription-vocabularies/${transcriptionVocabularyID}`, {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
    /**
     * Updates the details of a previously-created Transcription Vocabulary. Updates to
     * Transcription Vocabularies are allowed while associated live streams are active.
     * However, updates will not be applied to those streams while they are active.
     *
     * @example
     * ```ts
     * const transcriptionVocabulary =
     *   await client.video.transcriptionVocabularies.update(
     *     'TRANSCRIPTION_VOCABULARY_ID',
     *     {
     *       phrases: ['Mux', 'Live Stream', 'RTMP', 'Stream Key'],
     *       name: 'Mux API Vocabulary - Updated',
     *     },
     *   );
     * ```
     */
    update(transcriptionVocabularyID, body, options) {
        return this._client.put(path `/video/v1/transcription-vocabularies/${transcriptionVocabularyID}`, {
            body,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
    /**
     * List all Transcription Vocabularies.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const transcriptionVocabulary of client.video.transcriptionVocabularies.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/video/v1/transcription-vocabularies', (BasePage), { query, defaultBaseURL: 'https://api.mux.com', ...options });
    }
    /**
     * Deletes a Transcription Vocabulary. The Transcription Vocabulary's ID will be
     * disassociated from any live streams using it. Transcription Vocabularies can be
     * deleted while associated live streams are active. However, the words and phrases
     * in the deleted Transcription Vocabulary will remain attached to those streams
     * while they are active.
     *
     * @example
     * ```ts
     * await client.video.transcriptionVocabularies.delete(
     *   'TRANSCRIPTION_VOCABULARY_ID',
     * );
     * ```
     */
    delete(transcriptionVocabularyID, options) {
        return this._client.delete(path `/video/v1/transcription-vocabularies/${transcriptionVocabularyID}`, {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
            headers: buildHeaders([{ Accept: '*/*' }, options?.headers]),
        });
    }
}
