// @mux/mux-node/resources/video/uploads.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Uploads = void 0;
const resource_1 = require("../../core/resource.js");
const pagination_1 = require("../../core/pagination.js");
const path_1 = require("../../internal/utils/path.js");
/**
 * Direct upload allows you to push assets directly to Mux storage instead of needing to go through your own first. When you create a new direct upload, we'll give you back a signed URL for a Google Cloud Storage bucket. Their storage API is S3 compatible, so whatever tool you use to upload to either GCS or S3 should work, just remember you're probably uploading large video files and should [take advantage of things like resumable or multipart uploads](https://cloud.google.com/storage/docs/json_api/v1/how-tos/resumable-upload).
 *
 * Particularly for customers that deal with a lot of user-generated content, it's common to expect quite a few abandoned uploads. To keep those abandoned uploads from cluttering up your asset lists, we don't create an asset for you until the upload is complete. Once that asset is created, you can expect all of the normal asset-related webhooks.
 */
class Uploads extends resource_1.APIResource {
    /**
     * Creates a new direct upload, through which video content can be uploaded for
     * ingest to Mux.
     *
     * @example
     * ```ts
     * const upload = await client.video.uploads.create({
     *   cors_origin: 'https://example.com/',
     *   new_asset_settings: { playback_policies: ['public'] },
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/video/v1/uploads', {
            body,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
    /**
     * Fetches information about a single direct upload in the current environment.
     *
     * @example
     * ```ts
     * const upload = await client.video.uploads.retrieve(
     *   'abcd1234',
     * );
     * ```
     */
    retrieve(uploadID, options) {
        return this._client.get((0, path_1.path) `/video/v1/uploads/${uploadID}`, {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
    /**
     * Lists direct uploads in the current environment.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const upload of client.video.uploads.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/video/v1/uploads', (pagination_1.BasePage), {
            query,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        });
    }
    /**
     * Cancels a direct upload and marks it as cancelled. If a pending upload finishes
     * after this request, no asset will be created. This request will only succeed if
     * the upload is still in the `waiting` state.
     *
     * @example
     * ```ts
     * const upload = await client.video.uploads.cancel(
     *   'abcd1234',
     * );
     * ```
     */
    cancel(uploadID, options) {
        return this._client.put((0, path_1.path) `/video/v1/uploads/${uploadID}/cancel`, {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
}
exports.Uploads = Uploads;
