// @mux/mux-node/resources/video/video.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../core/resource.mjs";
import * as AssetsAPI from "./assets.mjs";
import { Assets, } from "./assets.mjs";
import * as DeliveryUsageAPI from "./delivery-usage.mjs";
import { DeliveryUsage, } from "./delivery-usage.mjs";
import * as DRMConfigurationsAPI from "./drm-configurations.mjs";
import { DRMConfigurations, } from "./drm-configurations.mjs";
import * as LiveStreamsAPI from "./live-streams.mjs";
import { LiveStreams, } from "./live-streams.mjs";
import * as PlaybackAPI from "./playback.mjs";
import { Playback, } from "./playback.mjs";
import * as PlaybackIdsAPI from "./playback-ids.mjs";
import { PlaybackIds } from "./playback-ids.mjs";
import * as PlaybackRestrictionsAPI from "./playback-restrictions.mjs";
import { PlaybackRestrictions, } from "./playback-restrictions.mjs";
import * as TranscriptionVocabulariesAPI from "./transcription-vocabularies.mjs";
import { TranscriptionVocabularies, } from "./transcription-vocabularies.mjs";
import * as UploadsAPI from "./uploads.mjs";
import { Uploads, } from "./uploads.mjs";
export class Video extends APIResource {
    constructor() {
        super(...arguments);
        this.assets = new AssetsAPI.Assets(this._client);
        this.deliveryUsage = new DeliveryUsageAPI.DeliveryUsage(this._client);
        this.liveStreams = new LiveStreamsAPI.LiveStreams(this._client);
        this.playbackIds = new PlaybackIdsAPI.PlaybackIds(this._client);
        this.playbackRestrictions = new PlaybackRestrictionsAPI.PlaybackRestrictions(this._client);
        this.transcriptionVocabularies = new TranscriptionVocabulariesAPI.TranscriptionVocabularies(this._client);
        this.uploads = new UploadsAPI.Uploads(this._client);
        this.drmConfigurations = new DRMConfigurationsAPI.DRMConfigurations(this._client);
        this.playback = new PlaybackAPI.Playback(this._client);
    }
}
Video.Assets = Assets;
Video.DeliveryUsage = DeliveryUsage;
Video.LiveStreams = LiveStreams;
Video.PlaybackIds = PlaybackIds;
Video.PlaybackRestrictions = PlaybackRestrictions;
Video.TranscriptionVocabularies = TranscriptionVocabularies;
Video.Uploads = Uploads;
Video.DRMConfigurations = DRMConfigurations;
Video.Playback = Playback;
