// @mux/mux-node/resources/video/video.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Video = void 0;
const tslib_1 = require("../../internal/tslib.js");
const resource_1 = require("../../core/resource.js");
const AssetsAPI = tslib_1.__importStar(require("./assets.js"));
const assets_1 = require("./assets.js");
const DeliveryUsageAPI = tslib_1.__importStar(require("./delivery-usage.js"));
const delivery_usage_1 = require("./delivery-usage.js");
const DRMConfigurationsAPI = tslib_1.__importStar(require("./drm-configurations.js"));
const drm_configurations_1 = require("./drm-configurations.js");
const LiveStreamsAPI = tslib_1.__importStar(require("./live-streams.js"));
const live_streams_1 = require("./live-streams.js");
const PlaybackAPI = tslib_1.__importStar(require("./playback.js"));
const playback_1 = require("./playback.js");
const PlaybackIdsAPI = tslib_1.__importStar(require("./playback-ids.js"));
const playback_ids_1 = require("./playback-ids.js");
const PlaybackRestrictionsAPI = tslib_1.__importStar(require("./playback-restrictions.js"));
const playback_restrictions_1 = require("./playback-restrictions.js");
const TranscriptionVocabulariesAPI = tslib_1.__importStar(require("./transcription-vocabularies.js"));
const transcription_vocabularies_1 = require("./transcription-vocabularies.js");
const UploadsAPI = tslib_1.__importStar(require("./uploads.js"));
const uploads_1 = require("./uploads.js");
class Video extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.assets = new AssetsAPI.Assets(this._client);
        this.deliveryUsage = new DeliveryUsageAPI.DeliveryUsage(this._client);
        this.liveStreams = new LiveStreamsAPI.LiveStreams(this._client);
        this.playbackIds = new PlaybackIdsAPI.PlaybackIds(this._client);
        this.playbackRestrictions = new PlaybackRestrictionsAPI.PlaybackRestrictions(this._client);
        this.transcriptionVocabularies = new TranscriptionVocabulariesAPI.TranscriptionVocabularies(this._client);
        this.uploads = new UploadsAPI.Uploads(this._client);
        this.drmConfigurations = new DRMConfigurationsAPI.DRMConfigurations(this._client);
        this.playback = new PlaybackAPI.Playback(this._client);
    }
}
exports.Video = Video;
Video.Assets = assets_1.Assets;
Video.DeliveryUsage = delivery_usage_1.DeliveryUsage;
Video.LiveStreams = live_streams_1.LiveStreams;
Video.PlaybackIds = playback_ids_1.PlaybackIds;
Video.PlaybackRestrictions = playback_restrictions_1.PlaybackRestrictions;
Video.TranscriptionVocabularies = transcription_vocabularies_1.TranscriptionVocabularies;
Video.Uploads = uploads_1.Uploads;
Video.DRMConfigurations = drm_configurations_1.DRMConfigurations;
Video.Playback = playback_1.Playback;
