// @mux/mux-node/resources/robots-preview/jobs/jobs.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Jobs = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const AskQuestionsAPI = tslib_1.__importStar(require("./ask-questions.js"));
const ask_questions_1 = require("./ask-questions.js");
const FindKeyMomentsAPI = tslib_1.__importStar(require("./find-key-moments.js"));
const find_key_moments_1 = require("./find-key-moments.js");
const GenerateChaptersAPI = tslib_1.__importStar(require("./generate-chapters.js"));
const generate_chapters_1 = require("./generate-chapters.js");
const ModerateAPI = tslib_1.__importStar(require("./moderate.js"));
const moderate_1 = require("./moderate.js");
const SummarizeAPI = tslib_1.__importStar(require("./summarize.js"));
const summarize_1 = require("./summarize.js");
const TranslateCaptionsAPI = tslib_1.__importStar(require("./translate-captions.js"));
const translate_captions_1 = require("./translate-captions.js");
const pagination_1 = require("../../../core/pagination.js");
const path_1 = require("../../../internal/utils/path.js");
/**
 * List and cancel jobs across all workflows.
 */
class Jobs extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.askQuestions = new AskQuestionsAPI.AskQuestions(this._client);
        this.generateChapters = new GenerateChaptersAPI.GenerateChapters(this._client);
        this.findKeyMoments = new FindKeyMomentsAPI.FindKeyMoments(this._client);
        this.moderate = new ModerateAPI.Moderate(this._client);
        this.summarize = new SummarizeAPI.Summarize(this._client);
        this.translateCaptions = new TranslateCaptionsAPI.TranslateCaptions(this._client);
    }
    /**
     * Returns a paginated list of Robots jobs, with optional filters for workflow,
     * status, and asset_id. Jobs are automatically deleted after 30 days.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const jobSummary of client.robotsPreview.jobs.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/robots/v0/jobs', (pagination_1.BasePage), {
            query,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        });
    }
    /**
     * Cancels a job that is currently pending or processing.
     *
     * @example
     * ```ts
     * const jobSummary = await client.robotsPreview.jobs.cancel(
     *   'rjob_lK9w2kI5J1',
     * );
     * ```
     */
    cancel(jobID, options) {
        return this._client.post((0, path_1.path) `/robots/v0/jobs/${jobID}/cancel`, {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
}
exports.Jobs = Jobs;
Jobs.AskQuestions = ask_questions_1.AskQuestions;
Jobs.GenerateChapters = generate_chapters_1.GenerateChapters;
Jobs.FindKeyMoments = find_key_moments_1.FindKeyMoments;
Jobs.Moderate = moderate_1.Moderate;
Jobs.Summarize = summarize_1.Summarize;
Jobs.TranslateCaptions = translate_captions_1.TranslateCaptions;
