// @mux/mux-node/resources/robots-preview/jobs/jobs.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as AskQuestionsAPI from "./ask-questions.mjs";
import { AskQuestions, } from "./ask-questions.mjs";
import * as FindKeyMomentsAPI from "./find-key-moments.mjs";
import { FindKeyMoments, } from "./find-key-moments.mjs";
import * as GenerateChaptersAPI from "./generate-chapters.mjs";
import { GenerateChapters, } from "./generate-chapters.mjs";
import * as ModerateAPI from "./moderate.mjs";
import { Moderate, } from "./moderate.mjs";
import * as SummarizeAPI from "./summarize.mjs";
import { Summarize, } from "./summarize.mjs";
import * as TranslateCaptionsAPI from "./translate-captions.mjs";
import { TranslateCaptions, } from "./translate-captions.mjs";
import { BasePage } from "../../../core/pagination.mjs";
import { path } from "../../../internal/utils/path.mjs";
/**
 * List and cancel jobs across all workflows.
 */
export class Jobs extends APIResource {
    constructor() {
        super(...arguments);
        this.askQuestions = new AskQuestionsAPI.AskQuestions(this._client);
        this.generateChapters = new GenerateChaptersAPI.GenerateChapters(this._client);
        this.findKeyMoments = new FindKeyMomentsAPI.FindKeyMoments(this._client);
        this.moderate = new ModerateAPI.Moderate(this._client);
        this.summarize = new SummarizeAPI.Summarize(this._client);
        this.translateCaptions = new TranslateCaptionsAPI.TranslateCaptions(this._client);
    }
    /**
     * Returns a paginated list of Robots jobs, with optional filters for workflow,
     * status, and asset_id. Jobs are automatically deleted after 30 days.
     *
     * @example
     * ```ts
     * // Automatically fetches more pages as needed.
     * for await (const jobSummary of client.robotsPreview.jobs.list()) {
     *   // ...
     * }
     * ```
     */
    list(query = {}, options) {
        return this._client.getAPIList('/robots/v0/jobs', (BasePage), {
            query,
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        });
    }
    /**
     * Cancels a job that is currently pending or processing.
     *
     * @example
     * ```ts
     * const jobSummary = await client.robotsPreview.jobs.cancel(
     *   'rjob_lK9w2kI5J1',
     * );
     * ```
     */
    cancel(jobID, options) {
        return this._client.post(path `/robots/v0/jobs/${jobID}/cancel`, {
            defaultBaseURL: 'https://api.mux.com',
            ...options,
        })._thenUnwrap((obj) => obj.data);
    }
}
Jobs.AskQuestions = AskQuestions;
Jobs.GenerateChapters = GenerateChapters;
Jobs.FindKeyMoments = FindKeyMoments;
Jobs.Moderate = Moderate;
Jobs.Summarize = Summarize;
Jobs.TranslateCaptions = TranslateCaptions;
