// @mux/mux-node/resources/jwt.mjs
import { APIResource } from "../core/resource.mjs";
import * as jwt from "../lib/jwt.mjs";
import { DataTypeClaim, isMuxJWTSignOptionsMultiple, TypeClaim, TypeToken, } from "../lib/jwt.mjs";
export class Jwt extends APIResource {
    /**
     * Creates a new token or tokens to be used with a signed Playback ID
     */
    async signPlaybackId(playbackId, config = {}) {
        if (isMuxJWTSignOptionsMultiple(config)) {
            return this.signPlaybackIdMultipleTypes(playbackId, config);
        }
        else {
            return this.signPlaybackIdSingleType(playbackId, config);
        }
    }
    async signPlaybackIdSingleType(playbackId, config) {
        const claim = TypeClaim[config.type ?? 'video'];
        if (!claim) {
            throw new Error(`Invalid signature type: ${config.type}; Expected one of ${Object.keys(TypeClaim)}`);
        }
        const tokenOptions = {
            keyid: jwt.getSigningKey(this._client, config),
            subject: playbackId,
            audience: claim,
            expiresIn: config.expiration ?? '7d',
            noTimestamp: true,
            algorithm: 'RS256',
        };
        return jwt.sign(config.params ?? {}, await jwt.getPrivateKey(this._client, config), tokenOptions);
    }
    async signPlaybackIdMultipleTypes(playbackId, config) {
        const tokens = {};
        for (const typeOption of config.type) {
            let type;
            let params;
            if (Array.isArray(typeOption)) {
                [type, params] = typeOption;
            }
            else {
                type = typeOption;
                params = undefined;
            }
            const singleConfig = {
                ...config,
                type,
                params: {
                    ...config.params,
                    ...params,
                },
            };
            const token = await this.signPlaybackIdSingleType(playbackId, singleConfig);
            const tokenKey = TypeToken[type];
            if (tokenKey) {
                tokens[tokenKey] = token;
            }
        }
        return tokens;
    }
    /**
     * Creates a new token for a license for playing back DRM'd video content
     */
    async signDrmLicense(playbackId, config = {}) {
        const claim = TypeClaim[config.type ?? 'drm_license'];
        if (!claim) {
            throw new Error(`Invalid signature type: ${config.type}; Expected one of ${Object.keys(TypeClaim)}`);
        }
        const tokenOptions = {
            keyid: jwt.getSigningKey(this._client, config),
            subject: playbackId,
            audience: claim,
            expiresIn: config.expiration ?? '7d',
            noTimestamp: true,
            algorithm: 'RS256',
        };
        return jwt.sign(config.params ?? {}, await jwt.getPrivateKey(this._client, config), tokenOptions);
    }
    /**
     * Creates a new token to be used with a signed statistics request
     */
    async signViewerCounts(id, config = {}) {
        const claim = DataTypeClaim[config.type ?? 'video'];
        if (!claim) {
            throw new Error(`Invalid signature type: ${config.type}; Expected one of ${Object.keys(DataTypeClaim)}`);
        }
        const tokenOptions = {
            keyid: jwt.getSigningKey(this._client, config),
            subject: id,
            audience: claim,
            expiresIn: config.expiration ?? '7d',
            noTimestamp: true,
            algorithm: 'RS256',
        };
        return jwt.sign(config.params ?? {}, await jwt.getPrivateKey(this._client, config), tokenOptions);
    }
}
