// @mux/mux-node/resources/jwt.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Jwt = void 0;
const tslib_1 = require("../internal/tslib.js");
const resource_1 = require("../core/resource.js");
const jwt = tslib_1.__importStar(require("../lib/jwt.js"));
const jwt_1 = require("../lib/jwt.js");
class Jwt extends resource_1.APIResource {
    /**
     * Creates a new token or tokens to be used with a signed Playback ID
     */
    async signPlaybackId(playbackId, config = {}) {
        if ((0, jwt_1.isMuxJWTSignOptionsMultiple)(config)) {
            return this.signPlaybackIdMultipleTypes(playbackId, config);
        }
        else {
            return this.signPlaybackIdSingleType(playbackId, config);
        }
    }
    async signPlaybackIdSingleType(playbackId, config) {
        const claim = jwt_1.TypeClaim[config.type ?? 'video'];
        if (!claim) {
            throw new Error(`Invalid signature type: ${config.type}; Expected one of ${Object.keys(jwt_1.TypeClaim)}`);
        }
        const tokenOptions = {
            keyid: jwt.getSigningKey(this._client, config),
            subject: playbackId,
            audience: claim,
            expiresIn: config.expiration ?? '7d',
            noTimestamp: true,
            algorithm: 'RS256',
        };
        return jwt.sign(config.params ?? {}, await jwt.getPrivateKey(this._client, config), tokenOptions);
    }
    async signPlaybackIdMultipleTypes(playbackId, config) {
        const tokens = {};
        for (const typeOption of config.type) {
            let type;
            let params;
            if (Array.isArray(typeOption)) {
                [type, params] = typeOption;
            }
            else {
                type = typeOption;
                params = undefined;
            }
            const singleConfig = {
                ...config,
                type,
                params: {
                    ...config.params,
                    ...params,
                },
            };
            const token = await this.signPlaybackIdSingleType(playbackId, singleConfig);
            const tokenKey = jwt_1.TypeToken[type];
            if (tokenKey) {
                tokens[tokenKey] = token;
            }
        }
        return tokens;
    }
    /**
     * Creates a new token for a license for playing back DRM'd video content
     */
    async signDrmLicense(playbackId, config = {}) {
        const claim = jwt_1.TypeClaim[config.type ?? 'drm_license'];
        if (!claim) {
            throw new Error(`Invalid signature type: ${config.type}; Expected one of ${Object.keys(jwt_1.TypeClaim)}`);
        }
        const tokenOptions = {
            keyid: jwt.getSigningKey(this._client, config),
            subject: playbackId,
            audience: claim,
            expiresIn: config.expiration ?? '7d',
            noTimestamp: true,
            algorithm: 'RS256',
        };
        return jwt.sign(config.params ?? {}, await jwt.getPrivateKey(this._client, config), tokenOptions);
    }
    /**
     * Creates a new token to be used with a signed statistics request
     */
    async signViewerCounts(id, config = {}) {
        const claim = jwt_1.DataTypeClaim[config.type ?? 'video'];
        if (!claim) {
            throw new Error(`Invalid signature type: ${config.type}; Expected one of ${Object.keys(jwt_1.DataTypeClaim)}`);
        }
        const tokenOptions = {
            keyid: jwt.getSigningKey(this._client, config),
            subject: id,
            audience: claim,
            expiresIn: config.expiration ?? '7d',
            noTimestamp: true,
            algorithm: 'RS256',
        };
        return jwt.sign(config.params ?? {}, await jwt.getPrivateKey(this._client, config), tokenOptions);
    }
}
exports.Jwt = Jwt;
