// @mux/mux-node/lib/jwt.mjs
import { isKeyLike, keyFormatErrorMessage, pkcs1to8, toPkcs8Pem, unwrapPem } from "./jwt-util.mjs";
export var TypeClaim;
(function (TypeClaim) {
    TypeClaim["video"] = "v";
    TypeClaim["thumbnail"] = "t";
    TypeClaim["gif"] = "g";
    TypeClaim["storyboard"] = "s";
    TypeClaim["stats"] = "playback_id";
    TypeClaim["drm_license"] = "d";
})(TypeClaim || (TypeClaim = {}));
export var TypeToken;
(function (TypeToken) {
    TypeToken["video"] = "playback-token";
    TypeToken["thumbnail"] = "thumbnail-token";
    TypeToken["storyboard"] = "storyboard-token";
    TypeToken["drm_license"] = "drm-token";
    TypeToken["gif"] = "gif-token";
    TypeToken["stats"] = "stats-token";
})(TypeToken || (TypeToken = {}));
export const isMuxJWTSignOptionsMultiple = (config) => Array.isArray(config.type);
export var DataTypeClaim;
(function (DataTypeClaim) {
    DataTypeClaim["video"] = "video_id";
    DataTypeClaim["asset"] = "asset_id";
    DataTypeClaim["playback"] = "playback_id";
    DataTypeClaim["live_stream"] = "live_stream_id";
})(DataTypeClaim || (DataTypeClaim = {}));
function base64url(data) {
    let str = '';
    for (let i = 0; i < data.length; i++) {
        str += String.fromCharCode(data[i]);
    }
    return btoa(str).replace(/\+/g, '-').replace(/\//g, '_').replace(/=/g, '');
}
function base64urlString(str) {
    return base64url(new TextEncoder().encode(str));
}
function parseTimespan(value) {
    if (typeof value === 'number')
        return value;
    const trimmed = value.trim();
    const match = /^(\d+)\s*(seconds?|secs?|s|minutes?|mins?|m|hours?|hrs?|h|days?|d|weeks?|w)$/.exec(trimmed);
    if (match) {
        const n = parseInt(match[1], 10);
        const unit = match[2];
        if (unit.startsWith('s'))
            return n;
        if (unit.startsWith('m'))
            return n * 60;
        if (unit.startsWith('h'))
            return n * 3600;
        if (unit.startsWith('d'))
            return n * 86400;
        if (unit.startsWith('w'))
            return n * 604800;
    }
    const num = Number(trimmed);
    if (!isNaN(num))
        return num;
    throw new Error(`Invalid time span: ${value}`);
}
async function importPKCS8(pem) {
    return globalThis.crypto.subtle.importKey('pkcs8', unwrapPem(pem), { name: 'RSASSA-PKCS1-v1_5', hash: { name: 'SHA-256' } }, false, ['sign']);
}
export async function sign(payload, secretOrPrivateKey, options) {
    const now = Math.floor(Date.now() / 1000);
    const header = { alg: options.algorithm || 'RS256', typ: 'JWT' };
    const claims = {
        ...payload,
        ...(options.keyid ? { kid: options.keyid } : {}),
    };
    if (options.issuer)
        claims['iss'] = options.issuer;
    if (options.subject)
        claims['sub'] = options.subject;
    if (options.audience != null)
        claims['aud'] = options.audience;
    if (!options.noTimestamp)
        claims['iat'] = now;
    if (options.notBefore != null)
        claims['nbf'] = now + parseTimespan(options.notBefore);
    if (options.expiresIn != null)
        claims['exp'] = now + parseTimespan(options.expiresIn);
    const key = secretOrPrivateKey instanceof Uint8Array ?
        await importPKCS8(toPkcs8Pem(pkcs1to8(secretOrPrivateKey)))
        : secretOrPrivateKey;
    const headerB64 = base64urlString(JSON.stringify(header));
    const payloadB64 = base64urlString(JSON.stringify(claims));
    const signingInput = `${headerB64}.${payloadB64}`;
    const signature = await globalThis.crypto.subtle.sign({ name: 'RSASSA-PKCS1-v1_5' }, key, new TextEncoder().encode(signingInput));
    return `${signingInput}.${base64url(new Uint8Array(signature))}`;
}
export function getSigningKey(mux, opts) {
    const keyId = opts.keyId || mux.jwtSigningKey;
    if (!keyId) {
        throw new Error('Signing key required; pass a keyId option to mux.jwt.sign*(), a jwtSigningKey option to new Mux(), or set the MUX_SIGNING_KEY environment variable');
    }
    return keyId;
}
export async function getPrivateKey(mux, opts) {
    let key = await getPrivateKeyHelper(mux, opts);
    if (typeof key === 'string') {
        if (key.startsWith('-----BEGIN RSA PRIVATE')) {
            key = toPkcs8Pem(pkcs1to8(unwrapPem(key)));
        }
        return await importPKCS8(key);
    }
    else if (key instanceof Uint8Array) {
        return await importPKCS8(toPkcs8Pem(pkcs1to8(key)));
    }
    else if (isKeyLike(key)) {
        return key;
    }
    throw new TypeError(keyFormatErrorMessage);
}
async function getPrivateKeyHelper(mux, opts) {
    let key;
    if (opts.keySecret) {
        key = opts.keySecret;
    }
    else if (opts.keyFilePath) {
        throw new Error(`keyFilePath is not supported in this environment`);
    }
    else if (mux.jwtPrivateKey) {
        key = mux.jwtPrivateKey;
    }
    if (isKeyLike(key))
        return key;
    if (typeof key === 'string') {
        key = key.trim();
        if (key.startsWith('-----BEGIN')) {
            return key;
        }
        try {
            key = atob(key);
            if (key.startsWith('-----BEGIN')) {
                return key;
            }
        }
        catch {
            // fallthrough
        }
        throw new TypeError(keyFormatErrorMessage);
    }
    throw new TypeError('Private key required; pass a keySecret option to mux.jwt.sign*(), a jwtPrivateKey option to new Mux(), or set the MUX_PRIVATE_KEY environment variable');
}
